src/DataModel/IResource.ts

```diff
@@ -0,0 +1,5 @@
+export interface IResource
+{
+    "@id": string;
+    "@type"?: Array<string>
```

Comment:
That's interesting. I've nothing against `@` but I think your intention was to make the representation serialization-agnostic, right? This seems clearly JSON-LD-inspired.